### PR TITLE
User Workflow Landing Page

### PR DESCRIPTION
**Pull request by Ellahinator**
Changes:

- After signing up, the user will be redirected to `/signin` rather than the home page
- A successful sign-in will redirect the user to `/settings` rather than the home page
- Add an error message to the redirect URL, `/signin?message=cannot-verify-otp` when email confirmation fails for some reason   
- sign up flow:
  1. Check if user exists
  2. If user exists, attempt to sign in with email and password
     - If sign in is successful redirect to /settings on client
     image
     - If sign in fails (wrong password) toast 'account exists' on client and redirect to /signin
     image

  3. If user does not exist, sign up with email and password and redirect to /signin
     image
 
- `Toaster` is defaulted to use light theme and now using `richColors`.
- Added success toasts on signups and update passwords. (these actions redirect the user to another page so the toast is there for clarity)
image
image

- Added GitHub OAuth to this PR 
  - old pr https://github.com/trypear/pear-landing-page/pull/38
image

TODO:
- [x] https://github.com/trypear/pear-landing-page/issues/104
- [x] implement reset password functionality
  - users who forget their password are now able to reset their password. upon reset they are redirected to `/settings`.

# MUST DO @Fryingpannn 
- [x] In https://supabase.com/dashboard/project/_/auth/templates under the _Reset Password_ tab, change the `href` to `{{ .SiteURL }}/auth/confirm?token_hash={{ .TokenHash }}&type=email&redirectUrl={{ .RedirectTo }}`. shoutout to this [reddit post](https://www.reddit.com/r/Supabase/comments/1c257cg/comment/l61xcpd/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button).

### CODE REVIEW

**Comment by Fryingpannn** [comment]
Could you add some description @Ellahinator 👀

File: app/(auth)/actions.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export async function signin(formData: FormData) {
   }
 
   revalidatePath("/", "layout");
-  redirect("/");
+  redirect("/settings");
```
Comment (Fryingpannn) [inline app/(auth)/actions.ts:27]: I think we keep it to redirect to home page even if they signed in for now. Cause our settings page does not have much info yet

File: app/(auth)/actions.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export async function signup(formData: FormData) {
   }
 
   revalidatePath("/", "layout");
-  redirect("/");
+  redirect("/signin");
```
Comment (Fryingpannn) [inline app/(auth)/actions.ts:51]: Does this redirect to sign in after user signs up?

File: app/(auth)/auth/confirm/route.ts [hunk 3]
```diff
@@ -28,6 +28,6 @@ export async function GET(request: NextRequest) {
   }
 
   // return the user to an error page with some instructions
-  redirectTo.pathname = '/auth/auth-code-error'
+  redirectTo.pathname = '/signin?message=Could not verify OTP'
```
Comment (Fryingpannn) [inline app/(auth)/auth/confirm/route.ts:31]:
```suggestion
  redirectTo.pathname = '/signin?message=cannot-verify-otp'
```

File: app/(auth)/auth/confirm/route.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export async function GET(request: NextRequest) {
   const { searchParams } = new URL(request.url)
   const token_hash = searchParams.get('token_hash')
   const type = searchParams.get('type') as EmailOtpType | null
-  const next = searchParams.get('next') ?? '/'
+  const next = searchParams.get('next') ?? '/settings'
```
Comment (Fryingpannn) [inline app/(auth)/auth/confirm/route.ts:10]: What does this do?

Comment (Ellahinator) [reply app/(auth)/auth/confirm/route.ts:10]: It redirects the user to `/settings` on an successful **oauth** sign in

File: app/(auth)/actions.ts (hunk 2 above)
Comment (Ellahinator) [reply app/(auth)/actions.ts:51]: Yes

File: app/(auth)/actions.ts (hunk 1 above)
Comment (Ellahinator) [reply app/(auth)/actions.ts:27]: Ok sounds good

**Comment by Ellahinator** [comment]
@Fryingpannn I think we should just wait to merge this once we have the settings page fully implemented. wdyt? I will make this a draft PR in the mean time.

File: app/(auth)/actions.ts (hunk 2 above)
Comment (Fryingpannn) [reply app/(auth)/actions.ts:51]: Hmm could we have the user signed in automatically after they signed up? If complicated then no need.

Comment (Ellahinator) [reply app/(auth)/actions.ts:51]: Yes, but that will require disabling email confirmation.

**Comment by Fryingpannn** [comment]
Merge conflicts 👀

**Comment by Fryingpannn** [comment]
MUSTDO for the redirect link change completed

**Review by Fryingpannn — APPROVED** [review]

**Pull request merged**